src/components/backend-ai-registry-list.ts

```diff
@@ -689,7 +690,7 @@ class BackendAIRegistryList extends BackendAIPage {
                       validationMessage="${_t('registry.PleaseSelectOption')}"
                       value="${this.registryList[this.selectedIndex]?.type || this.registryType}">
             ${this._registryTypes.map((item) => html`
-              <mwc-list-item value="${item}" ?selected="${item === 'docker'}">${item}</mwc-list-item>
+              <mwc-list-item value="${item}" ?selected="${this.editMode ? item === this.registryList[this.selectedIndex]?.type : item === 'docker'}">${item}</mwc-list-item>
```

Comment:
And If the line is too long horizontally, then perhaps you could add line breaks for better readability.